### PR TITLE
Change namespace to TightenCo and alias to Illuminate namespace

### PR DESCRIPTION
**Pull request by damiani**
Resolves https://github.com/tightenco/collect/issues/54

I agree that moving this to a `TightenCo` namespace makes sense, and that we need to provide a fallback `Illuminate` namespace for existing users. @adamwathan's PR (https://github.com/tightenco/collect/pull/55) does this by duplicating all the classes, but I think @RemiCollin's solution (https://github.com/tightenco/collect/issues/54#issuecomment-329608480) to alias the classes would be easier to maintain. 

I _believe_ this PR accomplishes that—it loads an `alias.php` file via composer, which takes care of it all. This works in my testing...but a second (or third or fourth) set of eyes on this would be appreciated, preferably by someone with more composer chops than I have 😆

### CODE REVIEW

**Comment by mattstauffer** [comment]
Can we rename TightenCo to Tightenco to be consistent with our other packages?

I like this way a lot. Let me give it a day for us to think of other potential problems.

**Comment by damiani** [comment]
No.

**Comment by damiani** [comment]
Oh, OK fine. :)

File: src/Tightenco/Support/alias.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    Tightenco\Support\Arr::class => Illuminate\Support\Arr::class,
+    Tightenco\Support\Collection::class => Illuminate\Support\Collection::class,
+    Tightenco\Support\HigherOrderCollectionProxy::class => Illuminate\Support\HigherOrderCollectionProxy::class,
+];
+
+foreach ($aliases as $tighten => $illuminate) {
+    if (! class_exists($illuminate) && ! interface_exists($illuminate) && ! trait_exists($illuminate)) {
+        class_alias($tighten, $illuminate);
+    }
+}
```
Comment (damiani) [inline src/Tightenco/Support/alias.php]: I'm wondering if we would be wiser to alias either _all_ classes, or _none_ of them if there's a single conflict. Theoretically, we could end up with a mixture of Laravel and Collect classes.

Comment (mattstauffer) [reply src/Tightenco/Support/alias.php]: I sort of doubt it, but I still like the idea of a single check instead of many.

**Comment by mathieutu** [comment]
Ok, I'm just seeing that : https://github.com/laravel/framework/commit/5caa88acde3d8a82fcbb9bbec8dde45dc49d4b7b

**Pull request merged**